microchip/common/modules.nix

```
@@ -0,0 +1,11 @@
+{ pkgs, lib, ... }: {
+  nixpkgs.overlays = [
+    (import ./overlay.nix)
+  ];
+
+  boot = {
+    kernelPackages = pkgs.linuxPackagesFor pkgs.linux-icicle-kit;
```

I would use callPackages here:

```suggestion
    kernelPackages = pkgs.linuxPackagesFor pkgs.callPackage ./bsp/linux-icicle-kit.nix { pkgs = final; boot = pkgs.callPackage ./path/to/the/uboot/uboot-icicle-kit {}; };;
```

THis also allows systems to work that do not use "nixpkgs" nixos settings because they apply the same nixpkgs instance to a fleet of machines (an optimization to scale up evaluation of many machines in CI or for deployment).

---

For now I want would like to go with classical way of packages extension for uniformity as we did for nxp boards:
https://github.com/NixOS/nixos-hardware/blob/master/nxp/common/modules.nix
Later we'll see how can we improve it.

---

Fair. Maybe someone else can merge it than. I dropped this overlay now as well: https://github.com/NixOS/nixos-hardware/pull/619/files

---

As per your suggestion, now I removed the extra overlay.